worf/validators.py

```diff
@@ -6,75 +6,89 @@
+    ]
+
     def _coerce_bool(self, value):
-        return {
-            "1": True,
-            "0": False,
-            "true": True,
-            "false": False,
-            True: True,
-            False: False,
-        }.get(str(value).lower().strip(), None)
-
-    def _validate_bundle_bool(self, key):
-        coerced = self._coerce_bool(self.bundle[key])
+        if isinstance(value, bool):
+            return value
+
+        return self.boolean_values.get(str(value).lower().strip(), None)
+
+    def _validate_boolean(self, key):
+        value = self.bundle[key]
+        coerced = self._coerce_bool(value)
+
         if not isinstance(coerced, bool):
-            msg = "Field {} accepts a boolean. Got {}, coerced to {}. Full bundle:\n\n{}".format(
-                snake_to_camel(key),
-                self.bundle[key],
-                coerced,
-                self.bundle,
+            raise ValidationError(
+                f"Field {snake_to_camel(key)} accepts a boolean, got {value}, coerced to {coerced}"
             )
-            raise HTTP400(msg)
 
         return coerced
 
-    def _validate_bundle_date_or_none(self, key):
-        if isinstance(self.bundle[key], str):
-            return datetime.strptime(self.bundle[key], "%Y-%m-%d")
+    def _validate_date(self, key):
+        value = self.bundle[key]
+
+        if not isinstance(value, str):
+            return None
 
-        return None
+        return datetime.strptime(value, "%Y-%m-%d")
 
-    def _validate_bundle_many_to_many(self, key):
-        if not isinstance(self.bundle[key], list):
-            msg = "Field {} accepts an array. Got {} {} instead.".format(
-                snake_to_camel(key),
-                type(self.bundle[key]),
-                self.bundle[key],
+    def _validate_many_to_many(self, key):
+        value = self.bundle[key]
+
+        if not isinstance(value, list):
+            raise ValidationError(
+                f"Field {snake_to_camel(key)} accepts an array, got {type(value)} {value}"
             )
-            raise HTTP400(msg)
 
         self.coerce_array_of_integers(key)
 
-    def _validate_bundle_str(self, key, max_len):
-        if not isinstance(self.bundle[key], str):
+    def _validate_string(self, key, max_length):
+        value = self.bundle[key]
+
+        if not isinstance(value, str):
             raise ValidationError(f"Field {snake_to_camel(key)} accepts string")
 
-        if max_len is not None and len(self.bundle[key]) > max_len:
-            raise HTTP400(
-                f"Field {snake_to_camel(key)} accepts a maximum of {max_len} characters"
+        if key not in self.secure_fields:
+            value = strip_tags(value).strip()
+
+        if max_length is not None and len(value) > max_length:
+            raise ValidationError(
+                f"Field {snake_to_camel(key)} accepts a maximum of {max_length} characters"
             )
 
-        return strip_tags(self.bundle[key])
+        return value
```

Comment:
☝🏼 I added a `.strip()` and a check for whether the key is in `secure_fields`, secure fields don't strip tags or trim.